core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the sidebar of the simple editor.
+ * editor.
+ */
+
+oppia.directive('simpleEditorSidebar', [function() {
+  return {
+    restrict: 'E',
+    templateUrl: 'simpleEditor/sidebar',
+    controller: [
+        '$scope', 'EditorModeService', 'SimpleEditorQuestionsDataService',
+        function($scope, EditorModeService, SimpleEditorQuestionsDataService) {
+      $scope.SUBFIELD_LABELS = [
+        'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
+
+      $scope.setEditorModeToFull = EditorModeService.setModeToFull;
+
+      $scope.$on('simpleEditorLoaded', function() {
+        $scope.questions = (
+            SimpleEditorQuestionsDataService.getQuestionsInOrder());
+      });
+      $scope.$on('updateSimpleEditorSidebar', function() {
```

Comment:
I think $on is better than $watch, because $watch looks at every little change (even though it may not require a sidebar update). WDYT?